packages/cardpay-reward-programs/tests/test_add_metadata.py

```diff
@@ -0,0 +1,53 @@
+import json
+from tempfile import TemporaryDirectory
+
+import pyarrow.parquet as pq
+from cardpay_reward_programs.payment_tree import PaymentTree
+from cardpay_reward_programs.utils import write_parquet_file
+from cloudpathlib import AnyPath
+
+
+def test_adds_metadata_to_payment_tree():
+    # Values don't matter here, but this is representative of nested data
+    metadata = {
```

Comment:
Do we want to group the metadata into several components 

Keep "run_parameters" in a separate key?? In case, we want any additive changes to metadata.

Reply:
Yeah the run_parameters are under their own key, so we can add more things if required. I partly did that because I think pandas and other tools tend to add some of their own metadata so I think this can help us avoid clashes in the future.